cmd/proxy/wsl_integration_linux.go

```diff
@@ -28,20 +31,35 @@ import (
 )
 
 var (
-	debug        bool
-	upstreamAddr string
-	listenAddr   string
+	debug            bool
+	upstreamAddr     string
+	listenAddr       string
+	keyFile          string
+	certFile         string
+	rootCA           string
+	upstreamKeyFile  string
+	upstreamCertFile string
 )
 
 const (
-	k8sAPI            = "https://192.168.1.2:6443"
-	defaultListenAddr = "127.0.0.1:6443"
+	k8sAPI               = "https://192.168.1.2:6443"
+	defaultListenAddr    = "127.0.0.1:6443"
+	kubeAPIServerKeyFile = "/var/lib/rancher/k3s/server/tls/serving-kube-apiserver.key"
+	kubeAPIServerCrtFile = "/var/lib/rancher/k3s/server/tls/serving-kube-apiserver.crt"
+	rootCACrtFile        = "/var/lib/rancher/k3s/server/tls/server-ca.crt"
+	clientAdminKeyFile   = "/var/lib/rancher/k3s/server/tls/client-admin.key"
+	clientAdminCrtFile   = "/var/lib/rancher/k3s/server/tls/client-admin.crt"
 )
 
 func main() {
 	flag.BoolVar(&debug, "debug", false, "enable additional debugging")
 	flag.StringVar(&upstreamAddr, "upstream-addr", k8sAPI, "The upstream server's address.")
 	flag.StringVar(&listenAddr, "listen-addr", defaultListenAddr, "The server's address in an IP:PORT format.")
+	flag.StringVar(&keyFile, "key-file", kubeAPIServerKeyFile, "TLS private key file for downstream connection")
+	flag.StringVar(&certFile, "cert-file", kubeAPIServerCrtFile, "TLS certificate file for downstream connection")
+	flag.StringVar(&rootCA, "root-ca-file", rootCACrtFile, "root ca file for upstream connection")
+	flag.StringVar(&upstreamKeyFile, "upstream-key-file", clientAdminKeyFile, "TLS private key file for upstream connection")
+	flag.StringVar(&upstreamCertFile, "upstream-cert-file", clientAdminCrtFile, "TLS certificate file for upstream connection")
```

Comment:
I think we should have some description text stating what downstream and upstream are, to make these understandable.  Possible need to override `flag.CommandLine.Usage` to have expanded description before calling `flag.CommandLine.PrintDefaults()`.

Reply:
In the context of the proxy, the downstream is always referred to as the incoming connection source (client, etc) into the proxy and the upstream is the server that the proxy is forwarding request to. Still, I will try to elaborate on this in the flag descriptions.